relayer/src/chain/runtime.rs

```diff
@@ -170,7 +170,10 @@ impl<C: Chain + Send + 'static> ChainRuntime<C> {
                                 .broadcast(Arc::new(event_batch))
                                 .map_err(Kind::channel)?;
                         },
-                        Err(e) => error!("received error via event bus: {}", e),
+                        Err(e) => {
+                            error!("received error via event bus: {}", e);
+                            return Err(Kind::Channel.into());
```

Comment:
Hmm I think it is by design that this match arm does not `return Err(..)` here but instead permits the loop to continue. For context, please see https://github.com/informalsystems/ibc-rs/pull/895 (and follow-up #903). The idea is that the chain runtime would continue polling via `select!` here even if the monitor's connection to the full node is down (or some other error appears in the `event_receiver`), and it's the job of the monitor to retry and eventually heal if the connection to the node is re-established.

It would be good if you can verify that your changes do not introduce a regression. Could you please just double-check that? The recipe for testing is in the description of https://github.com/informalsystems/ibc-rs/pull/903 (see "Tested with").

Reply:
Also, I was trying earlier to test but was not sure how to exercise the `Err` path here. Any tips?

Reply:
I checked the source code of crossbeam, and the only possible error is when the sender end of the channel is dropped, and so it is unrecoverable. We probably never encounter this in production, because if the channel is disconnected, it will also result in the flooding of error messages in the relayer log.

Also, if you check the other branches, each of them are called with `?` at the end. This means any error happening in those places will break the loop and return error. We should have another issue to address this by having some way to recover from any error inside the loop.

Reply:
Thank you for the findings & explanation!

I agree with you that the loop is not fault-tolerant anyway. I originally thought that returning in the `Err` path here could make the runtime (and the Hermes process in general) exit early (instead of waiting on the monitor to retry). But it seems there is a regression in master and that has nothing to do with the changes in this branch.

> We should have another issue to address this by having some way to recover from any error inside the loop.

I agree. If you know how to capture this in an issue, could you please handle that? We didn't notice these problems (yet), but it would be good to track this issue and add a general method for handling recovery.

Reply:
Sure, filed #1025 for that.